src/main/java/com/moneytransfer/aspect/IdempotentTransferAspect.java

```diff
@@ -0,0 +1,162 @@
+import com.moneytransfer.dto.NewTransferDto;
+import com.moneytransfer.entity.Request;
+import com.moneytransfer.entity.Transaction;
+import com.moneytransfer.enums.TransactionStatus;
+import com.moneytransfer.exceptions.MoneyTransferException;
+import com.moneytransfer.exceptions.RequestConflictException;
+import com.moneytransfer.exceptions.ResourceNotFoundException;
+import com.moneytransfer.repository.TransactionRepository;
+import com.moneytransfer.service.GetAccountService;
+import com.moneytransfer.service.RequestService;
+import lombok.RequiredArgsConstructor;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Aspect
+@Component
+@RequiredArgsConstructor
+public class IdempotentTransferAspect {
+    /**
+     * The GetAccount service.
+     */
+    private final GetAccountService getAccountService;
+    /**
+     * The Request service.
+     */
+    private final RequestService requestService;
+    /**
+     * The transaction repository.
+     */
+    private final TransactionRepository transactionRepository;
+
+    /**
+     * Processes the idempotent transfer request.
+     * @param proceedingJoinPoint
+     * @param idempotentTransferRequest
+     * @param requestId
+     * @param newTransferDto
+     * @return a (successful) Transaction
+     * @throws Throwable
+     */
+    @Around("@annotation(idempotentTransferRequest) && execution(* transfer(java.util.UUID, com.moneytransfer.dto.NewTransferDto,..)) && args(requestId, newTransferDto,..)")
+    @Transactional(isolation = Isolation.SERIALIZABLE)
+    public Transaction handleIdempotentTransferRequest(ProceedingJoinPoint proceedingJoinPoint, IdempotentTransferRequest idempotentTransferRequest, UUID requestId, NewTransferDto newTransferDto) throws Throwable {
+        return processRequest(requestId, newTransferDto, proceedingJoinPoint);
+    }
+
+    /**
+     * Processes an idempotent transfer request.
+     *
+     * @param requestId
+     * @param newTransferDto
+     * @param proceedingJoinPoint
+     * @return
+     * @throws Throwable
+     */
+    private Transaction processRequest(final UUID requestId, final NewTransferDto newTransferDto, final ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
+        try {
+            var request = getOrSubmitRequest(requestId, newTransferDto);
+            validateIdempotent(request, newTransferDto);
+            return switch (request.getRequestStatus()) {
+                case SUBMITTED -> processTransfer(request, proceedingJoinPoint);
+                case RESOLVED -> retrieveTransaction(request);
+            };
+        } catch (RuntimeException e) {
```

Comment:
Not sure what exception will reach that point.

Reply:
I added this try/catch assuming that any changes made to cache with @Cachable,@CachePut in a transactional context are reverted when a rollback occurs... I have to research this more